### DISCUSSION TITLE
Why is the QLayout component focusable?

### DISCUSSION THREAD

**Discussion by kazeshini178** [quasarframework/quasar#15064] [General - Components / Directives / etc]
Curious if there is a particular reason the `QLayout` component is focus-able with `tabindex="-1"` which seems to be explicitly added below
https://github.com/quasarframework/quasar/blob/8d6cf17fe9d4a937b60ce2ad12d41efcc01ced05/ui/src/components/layout/QLayout.js#L231-L236

Main reason for asking is that this caused some odd behaviour with another component we are using in our app as the `QLayout` would unexpectedly gain focus.

**Comment by pdanpdan** [comment]
To allow focusing so you can scroll the content using keyboard

**Reply by ibrainventures** [reply]
Hi @pdanpdan ,

i have / had a strange effect on chrome by the tabindex -1, 
after closing a dialog by esc-key, the complete layout was partly black surrounded..

You can easy test it, by going with 
-> chrome desktop on quasar.dev 
-> click on the white content area
-> and press the esc key .. the look changes:

![grafik](https://user-images.githubusercontent.com/8803626/213039429-a95f88d9-a216-4707-96bc-1550bd792c98.png)

normaly it should look

![grafik](https://user-images.githubusercontent.com/8803626/213039996-b6735eff-c3be-4525-ab73-c5b2568dff52.png)

Taking the tabindex away from the qlayout component, and added it by hand to the `index.template.html` on
the `<body>` the effect is away (also on the `<div id="q-app"` .. ).

What do you think is the best / better way ?

Greets
J.

**Reply by pdanpdan** [reply]
I would say it is enough to set the CSS for `.q-layout:focus` to `outline: none`

**Reply by ibrainventures** [reply]
To be a little more specific: 

Is the tabindex -1 needed for any framwork-core  feature / function / event-ing / method-ing?
(Beside on any aria- or comfort-style feature)

Reason: I want to work without the tabindex -1 inside the qlayout and qdialog components.. 
If you may know (as the father of this attrs code line :-) ) why you set this, may you
can remember the side-effect .. (with or without the tabindex -1 ..)

**Reply by ibrainventures** [reply]
after some testing the sideeffect on the qdialog without the tabindex are clear "seeable" .. no focusing by tabbing on elements.

so there will only the qlayout left for the question..

**Reply by pdanpdan** [reply]
Without focusable QLayout a lot of focus gets on the document body, and that behaves in strange ways across browsers (QDialog is one example)
Is there a specific reason for you not to have QLayout focusable (except visual glitch with outline that is fixable)?
